lms/djangoapps/courseware/tests/test_credit_requirements.py

```diff
@@ -133,6 +138,29 @@ def test_credit_requirements_not_eligible(self):
         )
         self.assertContains(response, "Verification Failed")
 
+    @ddt.data(CourseMode.HONOR, CourseMode.AUDIT, CourseMode.PROFESSIONAL)
+    def test_credit_requirements_non_credit_enrollment(self, enrollment_mode):
+        # Test the progress table is not displayed to the non credit students.
+        self.enrollment.mode = enrollment_mode
+        self.enrollment.save()  # pylint: disable=no-member
+
+        # Check the progress page display
+        response = self._get_progress_page()
+        self.assertNotContains(response, self.REQUIREMENT_HEADING)
+
+    @ddt.data(CourseMode.VERIFIED, CourseMode.CREDIT_MODE)
+    def test_credit_requirements_credit_verified_enrollment(self, enrollment_mode):
+        # Test the progress table is only displayed to the
+        # verified and credit students.
+        self.enrollment.mode = enrollment_mode
+        self.enrollment.save()  # pylint: disable=no-member
+
+        # Check the progress page display
+        response = self._get_progress_page()
+        # Expect that the requirements are displayed
+        self.assertContains(response, self.MIN_GRADE_REQ_DISPLAY)
+        self.assertContains(response, self.VERIFICATION_REQ_DISPLAY)
```

Comment:
As @jimabramson suggested, I'd encourage you to check for the presence of the `credit-eligibility` section. Doing so would make these tests more robust to text changes in the template.